File: workspace-graphx/src/main/scala/triangleCount.scala
import java.io._
import org.apache.spark._
import org.apache.spark.rdd.RDD
import org.apache.spark.graphx._
import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.graphx.lib._
import org.apache.spark.storage.StorageLevel

object triangleCount {
    def main(args: Array[String]): Unit = {
        
        // Create a Spark configuration and context
        val conf = new SparkConf()
            .setAppName("triangleCount")
            .setMaster("spark://master:7077")
        val sc = new SparkContext(conf)

        val path = "file:///home/user/data/web-Google.txt"

        // Load the edges as a graph
        val graph: Graph[Int, Int] = GraphLoader.edgeListFile(
            sc,
            path,
            edgeStorageLevel = StorageLevel.MEMORY_AND_DISK,
            vertexStorageLevel = StorageLevel.MEMORY_AND_DISK
        )

        // Compute the triangle count
        val triangleCount = TriangleCount.run(graph).vertices.map(_._2).reduce(_ + _) / 3

        // Write the output to a file
        val outputs = new File("/home/user/workspace-graphx/outputs/triangleCount.txt")
        val bw = new BufferedWriter(new FileWriter(outputs))
        bw.write(triangleCount.toString)
        bw.write("\n")
        bw.close()

        // Stop the Spark context
        sc.stop()
    }
}

File: workspace-flink/src/triangleCount/src/main/java/jar/triangleCount.java
package jar;

import java.io.File;
import java.io.FileWriter;
import java.io.BufferedWriter;
import org.apache.flink.graph.*;
import org.apache.flink.graph.library.*;
import org.apache.flink.types.NullValue;
import org.apache.flink.api.java.tuple.*;
import org.apache.flink.api.java.DataSet;
import org.apache.flink.api.common.typeinfo.Types;
import org.apache.flink.api.java.ExecutionEnvironment;

public class triangleCount {

	public static void main(String[] args) throws Exception {

		// Setup the execution environment
		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();

		String edgeListFilePath = "/home/user/data/web-Google.txt";

		// Load the edges as a DataSet of Tuples
		DataSet<Tuple2<Integer, Integer>> edges = env.readTextFile(edgeListFilePath)
			.filter(line -> !line.startsWith("#"))
			.map(line -> {
				String[] tokens = line.split("\\s+");
				return new Tuple2<Integer, Integer>(Integer.parseInt(tokens[0]), Integer.parseInt(tokens[1]));
			}).returns(Types.TUPLE(Types.INT, Types.INT));

		// Create the graph
		Graph<Integer, NullValue, NullValue> graph = Graph.fromTuple2DataSet(edges, env);

		// Compute the triangle count
		DataSet<Tuple3<Integer,Integer,Integer>> result = graph.run(new TriangleEnumerator<Integer, NullValue, NullValue>());

		long triangles = result.count();

		// Write the output to a file
		File outputs = new File("/home/user/workspace-flink/outputs/triangleCount.txt");
		BufferedWriter bw = new BufferedWriter(new FileWriter(outputs));
		bw.write(triangles + "\n");
		bw.close();
	}
}
